website/blog/2024-03-28-using-post-templates.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Using post templates
+description: Explanation on what post templates are and how to use them.
+slug: using-post-templates
+authors: obradovic
+date: 2024-03-28
+tags: [eightshift, boilerplate, templates]
+hide_table_of_contents: false
+---
+
+Sometimes you need your posts to follow a specific structure, or you simply want to provide a good starting point for the editors. This blog post will cover what post templates are, how to set them up, and what options they offer.
+<!--truncate-->
+
+A simple use case of post templates can be a review or a testimonial. You want each review to be a separate post and to use a single Quote block.
+
+If you don’t have the new post type registered, you can refer to one of the previous blog posts which explains how to [register Custom Post Types](/blog/using-cpts-and-taxonomies). Once the new post type is registered, you can set the `template_lock` argument.
+
+## Template locking options
+
+The best way to lock the template is by adding the `template_lock` argument into a `return` statement from the `getPostTypeArguments` method in your post type class.
+
+```php
+return [
+	// ...
+	'template_lock' => 'all',
+];
+```
+
+There are three ways of locking the template:
+
+- `all` - blocks can’t be inserted, removed, or moved around. Content can be edited for existing blocks
+- `insert` - you can’t insert or remove blocks, but you can move them around
+- `contentOnly` - locks everything, not even the content can be modified (?) - it may work differently on default Gutenberg blocks, but this is how it behaves with Eightshift blocks
```

Comment:
Let's maybe drop this one, as it's not the best documented, and for what we known, might change at any time 😄

Reply:
Agreed, testing it and researching about it was an experience to say the least 😂